### PR TITLE
Firestore updates

### PR DESCRIPTION
**Pull request by chdero**
Added fsmanager functions for writing and updating ratings, finding ratings from a userid and menuitemid, and getting a specific field from a document, given the docs id, and what field you want to retrieve. (Useful for getting a menu items average rating, or getting the rating a user left for a specific item, etc)

### CODE REVIEW

File: lib/services/fsmanager.dart [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+      return false;
+    }
+  }
+
+  Future deleteDocById(String collection, String id) async {
+    try {
+      await _firestore.collection(collection).document(id).delete();
+      return "deleted";
+    } catch (e) {
+      print("error in findDocById() : " + e.toString());
```
Comment (carteralbrecht) [inline lib/services/fsmanager.dart:35]: Found a typo here (deleteDocById())

File: lib/services/fsmanager.dart [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+class FirestoreManager {
+  final AuthService _authServ = AuthService();
+  final Firestore _firestore = Firestore.instance;
+
+  // variables for our collection names in firestore
+  final String userCollection = "userInfo";
+  final String ratingsCollection = "ratings";
+  final String menuItemCollection = "menuitems";
+
+  Future findDocById(String collection, String id) async {
```
Comment (carteralbrecht) [inline lib/services/fsmanager.dart:15]: Could we return the doc or call it like docExists() since we return a boolean?

Comment (chdero) [reply lib/services/fsmanager.dart:15]: yup we can return the DocumentSnapshot, i'll update it now

**Review by carteralbrecht — APPROVED** [review]
Goated

**Pull request merged**